src/req-wrap-inl.h

```diff
@@ -38,6 +39,119 @@ void ReqWrap<T>::Dispatched() {
   req_.data = this;
 }
 
+template <typename T>
+void ReqWrap<T>::Cancel() {
+  uv_cancel(reinterpret_cast<uv_req_t*>(&req_));
+}
+
+/* Below is dark template magic designed to invoke libuv functions that
```

Comment:
This is fairly standard C++, I'm not sure why it consists of magic but I like the explanation. I'm a little surprised this works on all Node's compile targets though.